### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by barney-s**
Update docs

- Add development guide
- Add quickstart guide
- Simplify Readme
- Move api table from readme to api.md
- Update contributing.md

### CODE REVIEW

**Comment by barney-s** [comment]
@nan-yu  FYI

**Comment by barney-s** [comment]
@nan-yu please review

**Comment by barney-s** [comment]
@ant31 would you please LGTM

File: docs/quickstart.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# fork https://github.com/kubernetes-sigs/application
+# clone
+GITHUBID=<githubid>
+git clone git@github.com:${GITHUBID}/application.git $GOPATH/src/sigs.k8s.io/kubebuilder
+```
+
+Run locally
+```bash
+# create local cluster
+make e2e-setup
```
Comment (nan-yu) [inline docs/quickstart.md]: Did you forget to check in the `e2e-setup` and `e2e-cleanup` goals? I couldn't find them in the Makefile. @barney-s

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -2,21 +2,23 @@
...
 
-This project uses and is built with [kubebuilder](https://github.com/kubernetes-sigs/kubebuilder). Kubebuilder does generates some of the code. Prior to submitting a pull request please check that you are not altering generated code and check if you need to regenerated code due to your change.
+## Changing API
 
-1. Make changes to the [Application CRD](pkg/apis/app/v1beta1/application_types.go).
-1. Add [tests](pkg/apis/app/v1beta1/application_types_test.go).
-1. Regenerate the generated code using `kubebuilder generate`.
-1. Update the [example](docs/examples/example.yaml)
+This project uses and is built with [kubebuilder](https://github.com/kubernetes-sigs/kubebuilder).
+To regenerate code after changes to the [Application CRD](pkg/apis/app/v1beta1/application_types.go), run `make generate`. Typically `make all` would take care of it. Make sure you add enough [tests](pkg/apis/app/v1beta1/application_types_test.go). Update the [example](docs/examples/example.yaml)
```
Comment (nan-yu) [inline CONTRIBUTING.md]: The path has been changed to `api/v1beta1/application_types.go`

Comment (nan-yu) [inline CONTRIBUTING.md]: new path: `api/v1beta1/application_types_test.go`

File: docs/develop.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+Fork [Application Repo](https://github.com/kubernetes-sigs/application).
+Then clone your fork locally.
+
+```bash
+mkdir -p $GOPATH/src/sigs.k8s.io
+cd $GOPATH/src/sigs.k8s.io
+
+GITHUBID=<githubid>
+git clone git@github.com:${GITHUBID}/application.git $GOPATH/src/sigs.k8s.io/kubebuilder
```
Comment (nan-yu) [inline docs/develop.md]: should be `application`

File: docs/develop.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+```
+
+### Cluster access
+For running e2e tests and development testing you need access to a cluster. You could create a cluster with your cloud provider and ensure the `kubeconfig` points to the cluster. 
+
+##### Local cluster
+For local testing you could create a `kind`.
```
Comment (nan-yu) [inline docs/develop.md]: provide a link: https://github.com/kubernetes-sigs/kind?

File: CONTRIBUTING.md (hunk 2 above)
Comment (nan-yu) [inline CONTRIBUTING.md]:
```
Update the [sample](config/samples/app_v1beta1_application.yaml) and [examples](docs/examples)
```

File: docs/quickstart.md (hunk 1 above)
Comment (barney-s) [reply docs/quickstart.md]: its in the other CL pending merge #159

File: CONTRIBUTING.md (hunk 2 above)
Comment (barney-s) [reply CONTRIBUTING.md]: Sorry i didnt get time to fix it. Its merged will send another CL.

Comment (barney-s) [reply CONTRIBUTING.md]: thanks will fix in a follow up issue.

Comment (barney-s) [reply CONTRIBUTING.md]: actually sample is removed.

File: docs/develop.md (hunk 4 above)
Comment (barney-s) [reply docs/develop.md]: ack

File: CONTRIBUTING.md (hunk 2 above)
Comment (barney-s) [reply CONTRIBUTING.md]: done.

**Pull request merged**